tests/v3_validation/cattlevalidationtest/core/test_webhook_scale_host.py

```diff
@@ -0,0 +1,919 @@
+
+    middle_host = host_scale_group[1]
+    last_host = host_scale_group[2]
+
+    # Checking the name of the hosts
+    base_host_name = base_host_name.split('.')[0]
+    base_host_prefix = base_host_name.rstrip('0123456789')
+    base_host_suffix = base_host_name.split(base_host_prefix)[1]
+
+    for host in host_scale_group:
+        if host["id"] == base_host["id"]:
+            continue
+        current_host_name = host["hostname"]
+        current_host_name = current_host_name.split('.')[0]
+        current_host_prefix = current_host_name.rstrip('0123456789')
+        current_host_suffix = current_host_name.split(current_host_prefix)[1]
+
+        assert current_host_prefix == base_host_prefix
+
+        if base_host_suffix != "":
+            assert string.atoi(current_host_suffix) == string.atoi(base_host_suffix) + 1
+        else:
+            assert current_host_suffix == "2"
+
+        base_host_suffix = current_host_suffix
+
+    # (1-2): Execute Webhook again and ensure the scale
+    # cannot be incremented beyond max
+    wh_resp = requests.post(webhook_up_url)
+    assert wh_resp.status_code == 400
+    json_resp = json.loads(wh_resp.content)
+    print "JSON response is:"
+    print json_resp
+
+    expected_response = "Error Cannot scale above provided max " \
+                        "scale value in executing driver for scaleHost"
+    assert json_resp['message'] == expected_response
+
+    # Delete the webhook
+    delete_webhook_verify(env.id, webhook_up_id)
+
+    # (2-1): Create webhook for scale down mostRecent hosts
+    data = {
+        "name": random_str(),
+        "driver": "scaleHost",
+        "scaleHostConfig": {
+            "action": "down",
+            "amount": 1,
+            "hostTemplateId": base_host.hostTemplateId,
+            "deleteOption": "mostRecent",
+            "min": 1,
+            "max": 4
+        }
+    }
+
+    resp = create_webhook(env.id, data)
+    assert resp.status_code == 200
+    assert resp.url is not None
+    json_resp = json.loads(resp.content)
+    webhook_mostRecent_url = json_resp["url"]
+    webhook_mostRecent_id = json_resp["id"]
+
+    print "Webhook is " + repr(webhook_mostRecent_url)
+    print "Id is " + repr(webhook_mostRecent_id)
+
+    # Execute Webhook and verify that the scale is decremented by
+    # the amount specified
+    wh_resp = requests.post(webhook_mostRecent_url)
+    assert wh_resp.status_code == 200
+
+    last_host = client.reload(last_host)
+    last_host = wait_for_condition(client,
+                                   last_host,
+                                   lambda x: x.state == 'removed',
+                                   lambda x: 'Host state is ' + x.state,
+                                   timeout=DEFAULT_MACHINE_TIMEOUT
+                                   )
+    assert last_host.state == 'removed'
+    assert base_host.state == 'active'
+    assert middle_host.state == 'active'
+
+    hosts = client.list_host()
```

Comment:
I used it before, but the test will error prone, I think it's better to check the state of the host